homeassistant/components/seventeentrack/sensor.py

```diff
@@ -111,242 +94,160 @@ async def async_setup_entry(
+            self._tracking_number, {}
         )
+        return package_data["extra"]
 
 
-class SeventeenTrackData:
-    """Define a data handler for 17track.net."""
+def remove_entity(hass: HomeAssistant, account_id: str, tracking_number: str) -> bool:
+    """Remove entity itself."""
+    reg = er.async_get(hass)
+    entity_id = reg.async_get_entity_id(
+        "sensor",
+        "seventeentrack",
+        UNIQUE_ID_TEMPLATE.format(account_id, tracking_number),
+    )
+    if entity_id:
+        reg.async_remove(entity_id)
+        return True
+    return False
 
-    def __init__(
-        self,
-        client,
-        async_add_entities,
-        scan_interval,
-        show_archived,
-        show_delivered,
-        timezone,
-    ) -> None:
-        """Initialize."""
-        self._async_add_entities = async_add_entities
-        self._client = client
-        self._scan_interval = scan_interval
-        self._show_archived = show_archived
-        self.account_id = client.profile.account_id
-        self.packages: dict[str, Package] = {}
-        self.show_delivered = show_delivered
-        self.timezone = timezone
-        self.summary: dict[str, int] = {}
-        self.async_update = Throttle(self._scan_interval)(self._async_update)
-        self.first_update = True
-
-    async def _async_update(self):
-        """Get updated data from 17track.net."""
-        entities: list[entity.Entity] = []
-
-        try:
-            packages = await self._client.profile.packages(
-                show_archived=self._show_archived, tz=self.timezone
-            )
-            _LOGGER.debug("New package data received: %s", packages)
-
-            new_packages = {p.tracking_number: p for p in packages}
-
-            to_add = set(new_packages) - set(self.packages)
-
-            _LOGGER.debug("Will add new tracking numbers: %s", to_add)
-            if to_add:
-                entities.extend(
-                    SeventeenTrackPackageSensor(self, new_packages[tracking_number])
-                    for tracking_number in to_add
-                )
 
-            self.packages = new_packages
-        except SeventeenTrackError as err:
-            _LOGGER.error("There was an error retrieving packages: %s", err)
-
-        try:
-            self.summary = await self._client.profile.summary(
-                show_archived=self._show_archived
-            )
-            _LOGGER.debug("New summary data received: %s", self.summary)
-
-            # creating summary sensors on first update
-            if self.first_update:
-                self.first_update = False
-                entities.extend(
-                    SeventeenTrackSummarySensor(self, status, quantity)
-                    for status, quantity in self.summary.items()
-                )
+def notify_delivered(hass: HomeAssistant, friendly_name: str, tracking_number: str):
+    """Notify when package is delivered."""
+    LOGGER.info("Package delivered: %s", tracking_number)
```

Comment:
Please don't use the info level of logging

Reply:
changed to debug